### PR TITLE
Add support for triu and tril

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/ops/tril.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+XlaOpVector Tril::Lower(LoweringContext* loctx) const {
+  xla::XlaOp input = loctx->GetOutputOp(operand(0));
+  xla::XlaOp output = BuildTril(input, diagonal_);
+  return ReturnOp(output, loctx);
+}
+
+std::string Tril::ToString() const {
+  std::stringstream ss;
+  ss << Node::ToString() << " diagonal=" << diagonal_;
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/tril.cpp]: , diagonal

File: torch_xla/csrc/ops/triu.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+XlaOpVector Triu::Lower(LoweringContext* loctx) const {
+  xla::XlaOp input = loctx->GetOutputOp(operand(0));
+  xla::XlaOp output = BuildTriu(input, diagonal_);
+  return ReturnOp(output, loctx);
+}
+
+std::string Triu::ToString() const {
+  std::stringstream ss;
+  ss << Node::ToString() << " diagonal=" << diagonal_;
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/triu.cpp]: Ditto

File: torch_xla/csrc/tensor.cpp [hunk 3]
```diff
@@ -929,6 +931,16 @@ XLATensor XLATensor::unsqueeze(const XLATensor& input, int dim) {
       input.GetDevice());
 }
 
+XLATensor XLATensor::triu(const XLATensor& input, int diagonal) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: int -> xla::int64

File: torch_xla/csrc/ops/tril.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/ops/tril.cpp]: Done.

File: torch_xla/csrc/ops/triu.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/ops/triu.cpp]: Done.

File: torch_xla/csrc/tensor.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: Done.

File: torch_xla/csrc/matrix.h [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "tensorflow/compiler/xla/client/xla_builder.h"
+#include "torch/csrc/jit/ir.h"
+
+namespace torch_xla {
+
+xla::XlaOp BuildTriu(const xla::XlaOp& input, int diagonal);
```
Comment (dlibenzi) [inline torch_xla/csrc/matrix.h]: I'd use int64 as well here, for uniformity.

**Pull request merged**